### PR TITLE
Bug Fix: JMS retrying does not work

### LINKED ISSUE #2053: JMS Retrying is not functioning.
### Description:

JMS Retrying is not functioning properly after moving the JMS connection-related logic to pkg directory.  Eventhough the reconnect happens the channel consuming has not been started after reconnect.

### Steps to reproduce:
Start docker-compose with apim pack.
restart the apim component
Try creating the some api/application 

### Affected Product Version:
current master branch

### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s): docker-compose , k8s 

---

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose
Fixes #2053.
In addition to that, the whole JMS related logic is moved to the common package. The publicly exposed method is,
InitJMSConnnection() which needs to be used to start the JMS connection.

Then onwards,
ThrottleDataChannel, NotificationChannel, KeyManagerChannel and RevokedTokenChannel could be used to fetch the JMS events.

### Issues

Fixes #2053

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/pkg/messaging/connection.go [hunk 1]
```diff
@@ -212,3 +231,98 @@ type amqpFailoverURL struct {
 	retryCount      int
 	connectionDelay int
 }
+
+func handleEvent(c *Consumer, key string) error {
+	var err error
+
+	logger.LoggerMsg.Debugf("got Connection, getting Channel for %s events", key)
```
Comment (Rajith90) [inline adapter/pkg/messaging/connection.go:238]: Can we make this a proper message. Here we do not get the the connection right, kind of misleading message it seems

File: adapter/pkg/messaging/connection.go [hunk 2]
```diff
@@ -212,3 +231,98 @@ type amqpFailoverURL struct {
...
+	queue, err := c.Channel.QueueDeclare(
+		"",    // name of the queue
+		false, // durable
+		true,  // delete when usused
+		false, // exclusive
+		false, // noWait
+		nil,   // arguments
+	)
+	if err != nil {
+		return fmt.Errorf("Queue Declare: %s", err)
```
Comment (Rajith90) [inline adapter/pkg/messaging/connection.go:267]:
```suggestion
		return fmt.Errorf("Error while declaring queue: %s", err)
```

File: adapter/pkg/messaging/connection.go [hunk 3]
```diff
@@ -212,3 +231,98 @@ type amqpFailoverURL struct {
...
+	}
+
+	logger.LoggerMsg.Debugf("declared Queue (%q %d messages, %d consumers), binding to Exchange (key %q)",
```
Comment (Rajith90) [inline adapter/pkg/messaging/connection.go:270]: Can we properly format this message. Once it builds seems it does not give a proper meaning

File: adapter/pkg/messaging/listener.go [hunk 4]
```diff
@@ -51,6 +63,22 @@ func StartConsumer(key string) (*Consumer) {
 	go func() {
 		c.reconnect(key)
 	}()
+	err := handleEvent(c, key)
+	if err != nil {
+		logger.LoggerMsg.Fatalf("%s", err)
+	}
+	if lifetime > 0 {
```
Comment (Rajith90) [inline adapter/pkg/messaging/listener.go:70]: It seems like lifetime is always zero. Can we please check this
https://github.com/wso2/product-microgateway/blob/1cde30dec45b60c3b9ac6b919b639e8b63260086/adapter/pkg/messaging/listener.go#L38

**Review by Rajith90 — CHANGES_REQUESTED** [review]

File: adapter/pkg/messaging/listener.go (hunk 4 above)
Comment (HiranyaKavishani) [reply adapter/pkg/messaging/listener.go:70]: We can either remove this or make it configurable.

**Review by pubudu538 — APPROVED** [review]

**Pull request merged**